src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
-package core.basesyntax;
-
-public class MyHashMap<K, V> implements MyMap<K, V> {
-
-    @Override
-    public void put(K key, V value) {
-
-    }
-
-    @Override
-    public V getValue(K key) {
-        return null;
-    }
-
-    @Override
-    public int getSize() {
-        return 0;
-    }
-}
+package core.basesyntax;
+
+public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int GROW_COEFFICIENT = 2;
+    private int treshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    private Node<K, V>[] table = new Node[DEFAULT_CAPACITY];
+    private int capacity = DEFAULT_CAPACITY;
+    private int size = 0;
+
+    @Override
+    public void put(K key, V value) {
+        int index = Math.abs(hash(key));
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (((node.key == null || key == null) && node.key == key)
+                    || ((node.key != null && key != null) && node.key.equals(key))) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K, V> newNode = new Node<>(index, key, value, table[index]);
+        table[index] = newNode;
+        size++;
+        if (size == treshold) {

```

Comment:
let's first check whether it is necessary to call the resize method and then add a new element